### DISCUSSION TITLE
Many small clusters while clustering in WNN space

### DISCUSSION THREAD

**Discussion by Florian411** [satijalab/seurat#5427] [Q&A]
Hello everyone,

I am currently working on a fairly large multimodal dataset of 314,336 cells with RNA and ADT data.
I realized that I have to set the "k" parameter very high (100 at least instead of 20) to avoid ending up with hundreds of clusters of size 2 or 3 in WNN space, while the default parameters work well as long as clustering happens in RNA and ADT space separately. 

Did someone experience something similar or has a good recommendation on how to handle such large data sets in multimodal space?

Thank you in advance and best regards,
Florian

**Comment by JovMaksimovic** [comment]
We have a dataset with >160,000 cells with RNA+ADT data and have observed the same behaviour when running WNN clustering. However, we didn't observe this behaviour when running the same analysis on data with < 20,000 cells. We will try the solution suggested by OP.

**Comment by lzj1769** [comment]
I have also observed similar results on a dataset with >90K cells from RNA+ATAC. A lot of clusters (>200) only have 2-3 cells after clustering based on wsnn graph, even with a low resolution (0.2). Not sure why this happened. In the end, I just removed them and only kept clusters with >100 cells.

**Comment by erflynn** [comment]
To fix this, I've decreased prune.snn to 1/20 instead of 1/15
